modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc

```diff
@@ -0,0 +1,260 @@
+<?php
+/**
+ * @file
+ * Ding DDBasic opening hours themes.
+ */
+
+/**
+ * A single library.
+ */
+function theme_ding_ddbasic_opening_hours_week($variables) {
+  $classes = implode(' ', $variables['classes_array']);
+  return '<div class="' . $classes . '">' . render($variables['table']) . '</div>';
+}
+
+/**
+ * Preprocessor for ding_ddbasic_opening_hours_week.
+ */
+function template_preprocess_ding_ddbasic_opening_hours_week(&$variables) {
+  $timespan = array(
+    strtotime('last monday', strtotime('+1 day')),
+    strtotime('next sunday', strtotime('-1 day')),
+  );
+  $today = strtotime('today');
+  $structured = array();
+  $categories = array();
+
+  if (!empty($variables['timespan'])) {
+    $timespan = $variables['timespan'];
+  }
+
+  // Get and sort all the instances in the given timespan.
+  $instances = opening_hours_instance_load_multiple(
+    array($variables['node']->nid),
+    date('Y-m-d', $timespan[0]),
+    date('Y-m-d', $timespan[1])
+  );
+
+  // Add closed days.
+  for ($day = $timespan[0]; $day <= $timespan[1]; $day += 86400) {
+    $date = date('Y-m-d', $day);
+    $closed = TRUE;
+    foreach ($instances as $instance) {
+      if ($instance->date == $date) {
+        $closed = FALSE;
+      }
+    }
+
+    if ($closed === TRUE) {
+      $instances[] = (object) array(
+        'instance_id' => -1,
+        'nid' => $variables['node']->nid,
+        'start_time' => '0',
+        'end_time' => '0',
+        'category_tid' => NULL,
+        'date' => $date,
+        'closed' => TRUE,
+        'notice' => t('Closed'),
+      );
+    }
+  }
+
+  usort($instances, 'ding_ddbasic_opening_hours_sort');
+  // Add true NULL to the categories.
+  if (in_array('_null', $variables['categories'])) {
+    $variables['categories'][NULL] = NULL;
+  }
+
+  // Insert the instances into the structured array.
+  foreach ($instances as $instance) {
+    if (!empty($variables['categories']) && !in_array($instance->category_tid, $variables['categories'])) {
+      continue;
+    }
+
+    $category_weight = ding_ddbasic_opening_hours_get_category_weight($instance->category_tid);
+    $time = strtotime($instance->date);
+    $day = format_date($time, 'custom', 'l');
+
+    if (!isset($structured[$day])) {
+      $structured[$day] = array(
+        'cols' => array(),
+        'extra' => array(),
+      );
+
+      if ($time == $today) {
+        $structured[$day]['extra']['class'] = array('today');
+      }
+    }
+
+    if (!empty($instance->notice)) {
+      $notice = '<span class="notice">' . $instance->notice . '</span>';
```

Comment:
Im still looking for some kind of answer here - Not necessarily a fix, but just an indictation what your thoughts are.